File: languages/js/src/entity/identity_v2/date_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

export class DateAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.DateAttributeDefinitionV2 {
    return identityEntitiesProto.DateAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.DateAttributeDefinitionV2
  ): DateAttributeDescriptor {
    return new DateAttributeDescriptor(r.displayName, r.id, r.description, r.required);
  }
}

File: languages/python/bloock/entity/identity_v2/date_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class DateAttributeDescriptor(AttributeDescriptor):
    @staticmethod
    def from_proto(a: proto.DateAttributeDefinitionV2) -> DateAttributeDescriptor:
        return DateAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
        )

    def to_proto(self) -> proto.DateAttributeDefinitionV2:
        return proto.DateAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
        )